rest-service/manager_rest/load_permissions.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import logging
+import os
+
+import yaml
+
+from manager_rest import config
+from manager_rest.flask_utils import setup_flask_app
+from manager_rest.storage.models_base import db
+
+
+def _update_roles(new_roles, keep):
+    existing_roles = {r.name: r for r in
+                      db.session.execute('select id, type, name from roles')}
+    logging.info('Found %d existing roles', len(existing_roles))
+    logging.debug('Found roles: %s', existing_roles)
+    new_roles = {r['name']: r for r in new_roles}
+    roles_to_drop = set(existing_roles) - set(new_roles)
+    roles_to_add = set(new_roles) - set(existing_roles)
+    for role in roles_to_add:
+        db.session.execute(
+            'insert into roles(name, type, description) '
+            'values (:name, :type, :description)',
+            {
+                'name': role['name'],
+                'description': role.get('description'),
+                'type': role.get('type', 'system_role')
+            }
+        )
+    logging.debug('Added roles: %s', roles_to_add)
+    logging.info('Inserted %d new roles', len(roles_to_add))
+    if not keep:
```

Comment:
I'd probably swap this around- if keep: logging.debug('not dropping'; else
Not a major thing, I just tend to prefer to make if statements positive where possible.